### ISSUE TITLE
Feature Request: Lines / Curves Annotations (Lane Lines)

### ISSUE THREAD

**Issue by d4l3k** [heartexlabs/label-studio#2563]
**Is your feature request related to a problem? Please describe.**

Label studio doesn't have a way to draw lines of different types on the images. This would be very useful for things such annotating driving footage with lane lines of different types, road edges, etc.

![2022-06-23-204553_1995x891_scrot](https://user-images.githubusercontent.com/909104/175458376-bb76320f-a219-4101-93d3-c7924358894a.png)

Example from CLRNet: https://arxiv.org/pdf/2203.10350v1.pdf

**Describe the solution you'd like**

A new labeling task that allows selecting different types of lines (i.e. solid white, solid yellow, dashed yellow, dashed white, etc) and drawing them on the image.

**Describe alternatives you've considered**

You can use segmentation for this but that produces a dense representation. A sparse representation for lane lines is often superior for planning tasks. I.e. Tesla is now using sparse representations for their lane graphs.

**Additional context**

**Comment by boother** [comment]
Yes, agree that Lines / Curves annotation is a very useful feature. I'm wondering is there any progress in its implementation ?

**Comment by alonks1234** [comment]
+1 for this feature

**Comment by ryouchinsa** [comment]
RectLabel is an offline image annotation tool for object detection and segmentation.
Although this is not an open source program, with RectLabel you can label the line segments for lane detection.
https://rectlabel.com/help#create_polygon

![line_segments](https://user-images.githubusercontent.com/1954306/214830504-96206813-7f0f-4e60-8b4f-7738d76aa0a8.jpg)

**Comment by bhavik494tatva** [comment]
Need to draw line on image.

**Comment by jonasweihing** [comment]
That feature would be really great for our student research group. Without polyline labeling it is unfortunately a dealbreaker.

**Comment by 9M6** [comment]
Would be awesome if we could get this.

**Comment by wangchenran** [comment]
This feature is great for me,  can we do it ? I'm wondering is there any progress in its implementation ?

**Comment by abdelDarwish510** [comment]
Any updates on this?

**Comment by makseq** [comment]
Unfortunately there are no updates :-(

**Comment by doantientai** [comment]
Yes, that would be helpful for training contour detection models! Still waiting in 2024

**Comment by jombooth** [comment]
/jira create
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10192331861)
> Jira issue [TRIAG-763](https://humansignal.atlassian.net/browse/TRIAG-763) is created

[TRIAG-763]: https://humansignal.atlassian.net/browse/TRIAG-763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
